plugins/devcycle/gatsby-ssr.js

```diff
@@ -1,19 +1,33 @@
 import React from 'react'
 import PropTypes from 'prop-types'
-import { withDVCProvider } from '@devcycle/devcycle-react-sdk'
-
-// this ensures that useVariable, used by useVariableValue, knows it's
-// in the DVCProvider, even if it's not initialized and generally useless
-// at this point
+import { useIsDevCycleInitialized, withDevCycleProvider } from '@devcycle/react-client-sdk'
 
 export const wrapRootElement = ({ element }, pluginOptions) => {
-    if (!pluginOptions.sdkKey) {
+    const sdkKey = pluginOptions.sdkKey.client
+    if (!sdkKey) {
         console.warn(
             'DevCycle SDK `sdkKey` is not configured. See https://github.com/devcycle/gatsby-plugin-devcycle#installation'
         )
     }
-    const App = () => <>{element}</>
-    const DevCycleApp = withDVCProvider(pluginOptions)(App)
+    const App = () => {
+        //to get correct value from useVariableValue, we need to wait for DevCycle to be initialized
+        const devcycleReady = useIsDevCycleInitialized()
+
+        if (!devcycleReady) {
+            return (
+                <div>
+                    <h1>DevCycle is not ready!</h1>
+                </div>
+            )
+        }
+        return <>{element}</>
+    }
```

Comment:
Is it ever not initialized at this point? I worry that if it's not, it would be unlikely that it ever would be at SSR, and it would return this html at all times making the sdk act solely client side.

Reply:
The best way to test this is to add some kind of dynamic value to this page so that it's forced to render on each and every page load, and then see if the variable values are returned within the html.

gatsby-node.js -> this is for static site generation, as in all the pages of your site can be built at build time, this is run by the build system at build time

gatsby-ssr.js -> this is for server side rendering, imagine a profile page that has specific user info, it needs to re-render via the server at each page load depending on the user, this is run by the server at request time

gatsby-client.js -> this is for client side, so what we're mostly familiar with, this is run by the browser at page load

So we want to make sure that gatsby-ssr.js has access to an initialized DevCycle SDK so that it doesn't just spit out a "Loading" equivalent page that the client-side then needs to update once the client-side bundle is loaded - since that defeats the purpose of server-side-rendering.

Make sense?